### ISSUE TITLE
Training with multiple input source

### ISSUE THREAD

**Issue by Lara192** [OpenNMT/OpenNMT-tf#325]
So I am trying to follow the documentation on how to use the multi_source_nmt.py model to train a multiple source language model. But I really am lost on how to execute this. I am quite new to all of this, so excuse me if it were obvious: 
This is the command I am using:
**onmt-main train_and_eval --model config\models\mutli_source_nmt.py --auto_config --config data.yml**

This is the error I am getting: ModuleNotFoundError: No module named 'mutli_source_nmt'

Or should I do something else? I noticed that this module is not defined in the catalog file of the models, but I thought that the --model argument is supposed to do the trick, like the documentation says...

As for the data.yml file:

  **train_features_file: data/en_src.txt
  train_labels_file: data/en_tgt.txt
  eval_features_file: data/en_val.txt
  eval_labels_file: raw-data/en_val_lab.txt
  source_vocabulary_1: data/en_vocab.txt 
  source_vocabulary_2: data/fr_vocab.txt
  target_vocabulary: data/de_vocab.txt** 

 Also from my understanding of the code (and bare with me), I have two vocab. source files for both languages, and one for the target vocab file for the target language, but I noticed that the second raw source file (for the second language) is not included, should I add it as well under train_features_file_2 for example, and same for eval_features_file...or is this the right way?

**Comment by guillaumekln** [comment]
> onmt-main train_and_eval --model config\models\mutli_source_nmt.py --auto_config --config data.yml

Is the path to the model file correct?

> should I add it as well under train_features_file_2 for example, and same for eval_features_file...or is this the right way?

The entry [Parallel inputs](http://opennmt.net/OpenNMT-tf/data.html#parallel-inputs) in the documentation should help. For both `train_features_file` and `eval_features_file` you should provide a list of files.

Also consider using `multi_source_transformer.py` which should give better performance.

**Comment by Lara192** [comment]
I think the path is correct: I have cloned the openNMT-tf repo, and created a variable (env) using virtualenv inside it,  and am accessing the model file from openNMT-tf folder .....\OpenNMT-tf> using the path I provided. So basically the env and config are in the same level.

**Comment by Lara192** [comment]
Traceback (most recent call last):
  File "~\appdata\local\programs\python\python36\Lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "~\appdata\local\programs\python\python36\Lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "~\OpenNMT-tf\pyenv\Scripts\onmt-main.exe\__main__.py", line 9, in <module>
  File "~\opennmt-tf\pyenv\lib\site-packages\opennmt\bin\main.py", line 143, in main
    serialize_model=is_chief)
  File "~\opennmt-tf\pyenv\lib\site-packages\opennmt\config.py", line 101, in load_model
    model = load_model_from_file(model_file)
  File "~\opennmt-tf\pyenv\lib\site-packages\opennmt\config.py", line 44, in load_model_from_file
    module = load_model_module(path)
  File "~\opennmt-tf\pyenv\lib\site-packages\opennmt\config.py", line 27, in load_model_module
    module = import_module(module_name)
  File "~\opennmt-tf\pyenv\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mutli_source_nmt'

**Comment by guillaumekln** [comment]
Is this Windows 10? Are you running the command via CMD?

**Comment by Lara192** [comment]
yes

**Comment by guillaumekln** [comment]
I will try to check if there is something specific to Windows. It's rarely used for this application.

In the meantime, you can try to update the global `PYTHONPATH` variable to include the directory `config\models`: https://stackoverflow.com/questions/3701646/how-to-add-to-the-pythonpath-in-windows

**Comment by Lara192** [comment]
I tried to update the variable, but it didn't work

**Comment by Lara192** [comment]
Hi,
So after a week long of trying to understand the problem, somehow, today I had the clear mind to spot my mistake: 

onmt-main train_and_eval --model config\models\**mutli**_source_nmt.py --auto_config --config data.yml

Yes, it is spelling error (multi). I don't know why I kept copying from the set of commands that I have prepared beforehand and never managed to detect this misspelling until today. 

It works perfectly on windows, no need for anything to be reviewed here. Can't wait to see the inference/translation results.

Thanks a lot for your quick responses.
